src/db/file.rs

```diff
@@ -276,6 +262,51 @@ pub fn add_path_into_database<P: AsRef<Path>>(conn: &Connection,
     file_list_to_json(file_list_with_mimes)
 }
 
+#[cfg(not(windows))]
+fn load_mime_data() -> Result<Cookie> {
+    let cookie = Cookie::open(flags::MIME_TYPE)?;
+    cookie.load::<&str>(&[])?;
+    Ok(cookie)
+}
+
+#[cfg(not(windows))]
+fn detect_mime(content: &Vec<u8>, file_path: &Path, cookie: &Cookie) -> Result<String> {
+    let mime = cookie.buffer(&content)?;
+
+    // magic is not specific enough sometimes
+    match mime.as_str() {
+        "text/plain" => {
+            match file_path.extension().and_then(OsStr::to_str) {
+                Some("md") => Ok("text/markdown".to_owned()),
+                Some("rs") => Ok("text/rust".to_owned()),
+                Some("markdown") => Ok("text/markdown".to_owned()),
+                Some("css") => Ok("text/css".to_owned()),
+                Some("toml") => Ok("text/x-toml".to_owned()),
+                Some("js") => Ok("application/javascript".to_owned()),
+                Some("json") => Ok("application/json".to_owned()),
+                _ => Ok(mime.to_owned())
+            }
+        }
+        "text/troff" => {
+            match file_path.extension().and_then(OsStr::to_str) {
+                Some("css") => Ok("text/css".to_owned()),
+                _ => Ok(mime.to_owned())
+            }
+        }
+        _ => Ok(mime.to_owned())
```

Comment:
I see `to_owned()` a lot here, is there any way you could factor that out to only come at the end of the function? Might make it easier to return a borrow in the future.